packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -198,6 +199,120 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  bool _shiftIsDown = false;
```

Comment:
Won't these have the same issue that shift has?

Reply:
What issue is that? I think it should be fine. I check for the shift key being pressed with every key press so this variable should always represent the state of the shift key.

Reply:
Oh I see that now, perfect. Do you still need to track all of the keystate here if it is in Keyboard too?

Reply:
Yah, wait I don't because now I'm updating the value for shift with every key press.